File: Chapter4/Python/Example3.py
# Chapter 4 - Poisson Distribution
# Example 3: Visualization of Poisson(7)

import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import poisson

# Parameter
lam = 7

# Values (0 to 20 covers most of the distribution)
x = np.arange(0, 21)

# Calculate probabilities
pmf = poisson.pmf(x, lam)

# Create plot
plt.figure(figsize=(10, 6))
plt.bar(x, pmf, color='darkgreen', alpha=0.7, edgecolor='black')
plt.xlabel('Number of Events (k)', fontsize=12)
plt.ylabel('Probability P(X = k)', fontsize=12)
plt.title(f'Poisson Distribution with lambda = {lam}', fontsize=14)
plt.grid(axis='y', alpha=0.3)
plt.xticks(x)

# Mark mean value (= lambda)
plt.axvline(lam, color='red', linestyle='--',
            label=f'Mean = lambda = {lam}')
plt.legend()
plt.tight_layout()
plt.show()

File: Chapter4/R/Example3.R
# Chapter 4 - Poisson Distribution
# Example 3: Visualization of Poisson(7)

# Parameter
lambda <- 7

# Values (0 to 20 covers most of the distribution)
x <- 0:20

# Calculate probabilities
pmf <- dpois(x, lambda)

# Create plot
barplot(pmf, names.arg = x, col = 'darkgreen',
        xlab   = 'Number of Events (k)',
        ylab   = 'Probability P(X = k)',
        main   = paste0('Poisson Distribution with lambda = ', lambda),
        border = 'black')

# Mark mean value (= lambda)
abline(v = lambda + 0.5, col = 'red', lwd = 2, lty = 2)
legend('topright',
       legend = paste('Mean = lambda =', lambda),
       col = 'red', lty = 2, lwd = 2)
